src/Framework/ExecutionOrderDependency.php

```diff
@@ -56,9 +56,9 @@ public static function createFromDependsAnnotation(string $className, string $an
     }
 
     /**
-     * @param array<ExecutionOrderDependency> $dependencies
+     * @param list<ExecutionOrderDependency> $dependencies
      *
-     * @return array<ExecutionOrderDependency>
+     * @return ExecutionOrderDependency[]
```

Comment:
This is the only [] notation in this PR; is that intentional?

Reply:
Yes. It was the quickest and cleanest way to shut Psalm up regarding the list/array juggling around `filterInvalid()`. This was the main discussion with @jaapio this morning.

I'm all up for cleaner ways of stitching this back together!

Reply:
How about this, then:

```diff
diff --git a/src/Framework/ExecutionOrderDependency.php b/src/Framework/ExecutionOrderDependency.php
index fda9ada02..8f39cf567 100644
--- a/src/Framework/ExecutionOrderDependency.php
+++ b/src/Framework/ExecutionOrderDependency.php
@@ -58,13 +58,13 @@ public static function createFromDependsAnnotation(string $className, string $an
     /**
      * @param list<ExecutionOrderDependency> $dependencies
      *
-     * @return ExecutionOrderDependency[]
+     * @return list<ExecutionOrderDependency>
      */
     public static function filterInvalid(array $dependencies): array
     {
-        return array_filter($dependencies, function (self $d) {
+        return array_values(array_filter($dependencies, function (self $d) {
             return $d->isValid();
-        });
+        }));
     }

     /**
```

Reply:
Thank you, fixed :)